reconcile/utils/unleash.py

```diff
@@ -1,117 +1,94 @@
 import os
-import tempfile
-import shutil
 import threading
+from typing import Mapping, Optional, Any
 
-from UnleashClient import UnleashClient
+from UnleashClient import UnleashClient, BaseCache
 from UnleashClient import strategies
+from UnleashClient.strategies import Strategy
 
-from reconcile.utils.defer import defer
 from reconcile.utils.helpers import toggle_logger
 
-log_lock = threading.Lock()
+client_local = threading.local()
```

Comment:
So I understand correctly that there will be one unleash client per thread?

Reply:
Correct, I added a test to make that clearer.

Reply:
What is the reason to have one per thread?

Reply:
I kept it the way it was. The previous implementation had the same behavior but destroyed the client after every request. 

Let me test and read up if the client is thread safe.

Reply:
thank you for taking additional time looking into this ❤️